src/demo/components/ExampleCard/ExampleCard.tsx

```diff
@@ -42,9 +42,9 @@ export class ExampleCard extends React.Component<IExampleCardProps, IExampleCard
         <div className='ExampleCard-header'>
           <span className='ExampleCard-title ms-font-l'>{ title }</span>
           <span className='ExampleCard-toggleButtons ms-font-l'>
-            <Button buttonType={ ButtonType.primary } onClick={ this._onShowExampleClick }>{ isExampleShown ? 'Hide example' : 'Show example' }</Button>
+            <Button buttonType={ ButtonType.primary } rootProps={ { onClick: this._onShowExampleClick } }>{ isExampleShown ? 'Hide example' : 'Show example' }</Button>
```

Comment:
When i read this, i feel confused. Specifically for button. Button does not have a label or wrapper div. It's just a button. So rootProps is ambiguous. It makes me think there is a higher level wrapper around the button.

I expect this to work:

```typescript
<Button onClick={ stuff }>Hi</Button>
```

I think we should explicitly support the common props on the component props, and use the mixin "fooProps" values only for supporting exceptional uncommon mixin cases.

Here's another example:

```typescript
<Checkbox className='foo' label='' checked={ true } onChange={ this._onCheckboxChange } />
```
With the proposed pattern might look like this (keep in mind, the onChange we provide is an extension of the input box's onChange):

```typescript
<Checkbox label='' onChange={ this._onCheckboxChange } inputProps={ { className: 'foo', checked: true } }
```
This feels super weird! Doesn't it?

I believe we should consider using this pattern for exceptional 'mixin' scenarios that are uncommon. For the common things, we pass them through. So:

```typescript
<Checkbox 
  className='foo'
  checked={ true } 
  onChange={ this._onCheckboxChange } 
  inputProps={ { style: { background: 'red' } } } />
```